astropy/time/core.py

```diff
@@ -582,10 +582,10 @@ def _get_time_fmt(
         """
         if format is None:
             # If val and val2 broadcasted shape is (0,) (i.e. empty array input) then we
-            # cannot guess format from the input values.  Instead use the default
-            # format.
+            # cannot guess format from the input values. But a quantity is fine (as
+            # long as it has time units, but that will be checked later).
             empty_array = val.size == 0 and (val2 is None or val2.size == 0)
-            if empty_array or np.all(mask):
+            if (empty_array or np.all(mask)) and not isinstance(val, u.Quantity):
```

Comment:
Put the `not isinstance(...)` condition first to not do the potentially expensive `all(mask)` when it's not needed?

Reply:
I had wondered about it but felt quantity input was less likely, but thinking again, the `isinstance` is really negligible compared to everything else while going over the full mask may not be. Anyway, done now.